src/pages/game/game.page.tsx

```diff
@@ -9,12 +13,45 @@ import { useGameStore } from "@/stores/game.store.ts";
 import styles from "./game.module.css";
 
 export default function GamePage(): ReactNode {
+  const navigate = useNavigate();
+
   const username = localStorage.getItem("username");
 
   const score = useGameStore((state) => state.score);
+  const gameState = useGameStore((state) => state.gameState);
 
   const masterRef = useRef<GameMasterService>(null);
   const canvasRef = useRef<HTMLCanvasElement>(null);
+  const modalRef = useRef<HTMLDialogElement>(null);
+
+  const openButtonClickHandler = () => {
+    modalRef.current?.showModal();
+  };
+
+  const closeButtonClickHandler = () => {
+    modalRef.current?.close();
+  };
+
+  const resetGameStates = () => {
+    useGameStore.getState().run();
+    useGameStore.getState().resetScore();
+  };
+
+  const restartGame = () => {
+    if (masterRef.current && canvasRef.current) {
```

Comment:
اول نقیضش رو چک کن که الکی کل کد Indent نخوره

```suggestion
    if (!masterRef.current || !canvasRef.current) {
```